floodviz/hydrograph_utils.py

```diff
@@ -6,44 +6,38 @@ def parse_hydrodata(jdata):
     
     """ 
     Parses json Hydrodata from NWIS webservice
-    and formats for NVD3 charting library. Upon failure, this will
+    and formats for D3 charting library. Upon failure, this will
     return empty data list.
 
     ARGS: 
         jdata (list of one dictonary) - json object in a list which contains 
         time series data for all sites listed in config.py SITE_IDs.
     
     RETURNS:
-        All series data correctly formated for NVD3 charting library. 
-        This is also a list of python dictonaries where each dictionary 
-        corresponds to a different site.   
+        All series data correctly formated for D3.
+        This is also a list of python dictonaries.
 
     """
     all_series_data = []
     
     if jdata is not None:
-        for idx, site in enumerate(jdata):
+        for site in jdata:
 
-            site_name = site.get('sourceInfo').get('siteName')
-            if site_name is None:
-                site_name = 'NaN'    
-            all_series_data.append({'key': site_name, 'values': []})
+            site_name = site['sourceInfo']['siteName']
+            site_id = site['sourceInfo']['siteCode'][0]['value']
 
-            # Fill new data for this series
-            for idx2, obj in enumerate(site['values'][0]['value']):
+            # Fill data for this series
+            for obj in site['values'][0]['value']:
                 value = obj['value']
                 dt = obj['dateTime']
```

Comment:
In the future, we should consider a utility function that takes an NWIS data string and returns a python datetime object. No changes needed right now.